bin/oref0-ns-autotune.sh

```diff
@@ -0,0 +1,173 @@
+cd $directory && mkdir -p autotune
+cp settings/pumpprofile.json autotune/profile.pump.json
+# If a previous valid settings/autotune.json exists, use that; otherwise start from settings/profile.json
+cp settings/autotune.json autotune/profile.json && cat autotune/profile.json | json | grep -q start || cp autotune/profile.pump.json autotune/profile.json
+cd autotune
+# TODO: Need to think through what to remove in the autotune folder...
+
+# Pull Nightscout Data
+echo "Grabbing NIGHTSCOUT treatments.json for date range..."
+
+# Get Nightscout carb and insulin Treatments
+curl "$NIGHTSCOUT_HOST/api/v1/treatments.json?find\[created_at\]\[\$gte\]=`date -d $START_DATE -Iminutes`&\[\$lte\]=`date -d $END_DATE -Iminutes`" > ns-treatments.json
+
+# Build date list for autotune iteration
+date_list=()
+date=$START_DATE; 
+while :
+do 
+  date_list+=( "$date" )
+  if [ $date != "$END_DATE" ]; then 
+    date="$(date --date="$date + 1 days" +%Y-%m-%d)"; 
+  else 
+    break
+  fi
+done
+
+echo "Grabbing NIGHTSCOUT entries/sgv.json for date range..."
+
+# Get Nightscout BG (sgv.json) Entries
+for i in "${date_list[@]}"
+do 
+  curl "$NIGHTSCOUT_HOST/api/v1/entries/sgv.json?find\[date\]\[\$gte\]=`(date -d $i +%s | tr -d '\n'; echo 000)`&find\[date\]\[\$lte\]=`(date --date="$i +1 days" +%s | tr -d '\n'; echo 000)`&count=1000" > ns-entries.$i.json
+done
+
+echo "Running $NUMBER_OF_RUNS runs from $START_DATE to $END_DATE"
+sleep 2
+
+# Do iterative runs over date range, save autotune.json (prepped data) and input/output 
+# profile.json
+# Loop 1: Run 1 to Number of Runs specified by user or by default (1)
+for run_number in $(seq 1 $NUMBER_OF_RUNS)
+do
+  # Loop 2: Iterate through Date Range
+  for i in "${date_list[@]}"
+  do
+    cp profile.json profile.$run_number.$i.json
+    # Autotune Prep (required args, <pumphistory.json> <profile.json> <glucose.json>), output prepped glucose 
+    # data or <autotune/glucose.json> below
+    echo "~/src/oref0/bin/oref0-autotune-prep.js ns-treatments.json profile.json ns-entries.$i.json > autotune.$run_number.$i.json"
+    ~/src/oref0/bin/oref0-autotune-prep.js ns-treatments.json profile.json ns-entries.$i.json > autotune.$run_number.$i.json
+    
+    # Autotune  (required args, <autotune/glucose.json> <autotune/autotune.json> <settings/profile.json>), 
+    # output autotuned profile or what will be used as <autotune/autotune.json> in the next iteration
+    echo "~/src/oref0/bin/oref0-autotune.js autotune.$run_number.$i.json profile.json profile.pump.json > newprofile.$run_number.$i.json"
+    ~/src/oref0/bin/oref0-autotune.js autotune.$run_number.$i.json profile.json profile.pump.json > newprofile.$run_number.$i.json
+    
+    # Copy tuned profile produced by autotune to profile.json for use with next day of data
+    cp newprofile.$run_number.$i.json profile.json
+
+  done # End Date Range Iteration
+done # End Number of Runs Loop
+
```

Comment:
We could do something like:
```
date="$(date --date="$date + 1 days" +%Y-%m-%d-%H-%M)"; 
cp profile.autotune.json profile.autotune.$date.json 
```
here to save the generated autotune profile on disk for later reference/backup.
Even better will be to store the profile in Nightscout, see https://github.com/openaps/oref0/pull/301